pkg/provisioner/ironic/ironic.go

```diff
@@ -1295,3 +1323,28 @@ func (p *ironicProvisioner) PowerOff() (result provisioner.Result, err error) {
 
 	return result, nil
 }
+
+// getImageChecksum parses the checksum and returns image_hash and image_value
+// e.g. "md5:458950b8cbebad6d437bd35d4d2bf66a" returns "md5" and "458950b8cbebad6d437bd35d4d2bf66a"
+// "458950b8cbebad6d437bd35d4d2bf66a" returns a default algo: "md5", and "458950b8cbebad6d437bd35d4d2bf66a"
+// "sha256:http://172.16.1.100/images/myOSv1/myOS.qcow2.sha256sum" returns "sha256" and "http://172.16.1.100/images/myOSv1/myOS.qcow2.sha256sum"
+// "http://172.16.1.100/images/myOSv1/myOS.qcow2.md5sum" returns "md5" and "http://172.16.1.100/images/myOSv1/myOS.qcow2.md5sum"
+
+func getImageChecksum(checksum string) (string, string) {
```

Comment:
I suddenly wonder if putting this in the Ironic provisioner is a mistake. It effectively makes this field a pass-through for Ironic, which is the thing we're trying to avoid by having a Provisioner interface.
Potentially this could be a method of v1alpha1.Image, that returns the algorithm from a defined list and either the hash value or a URL (and it should probably be explicit about which, since it's likely only Ironic that doesn't care).

Reply:
If we add an extra filed to specify image_hash_algo along with image_hash_value, this will become moot.